File: app/src/main/java/com/herba/sdk/jetpackexample/roomRecyclerview/NoteDao.java
package com.herba.sdk.jetpackexample.roomRecyclerview;

import android.service.voice.VoiceInteractionService;

import java.util.List;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

@Dao
public interface NoteDao {

    @Insert
    void insert(Note note);

    @Query("SELECT * FROM notes")
    LiveData<List<Note>> getAllNotes();

    @Query("SELECT * FROM notes WHERE id =:noteId")
    LiveData<Note> getNotes(String noteId);

    @Update
    void update(Note note);



}

File: app-kotlin/src/main/java/com/example/myapplication/roomRecyclerview/utils/NoteDao.kt
package com.example.myapplication.roomRecyclerview.utils

import androidx.lifecycle.LiveData
import androidx.room.*

@Dao
interface NoteDao {

    @get:Query("SELECT * FROM notes ORDER BY appLabel ASC")
    val allApps: LiveData<List<Note>>

    @Insert
    fun insert(note: Note)

    @Query("SELECT * FROM notes WHERE appPackage =:appPackage")
    fun getApp(appPackage: String): LiveData<Note>

    @Query("SELECT COUNT(*) FROM notes WHERE appPackage =:appPackage")
    fun getAppCount(appPackage: String): Int

    @Update
    fun update(note: Note)

    @Delete
    fun delete(note: Note)

}